### PR TITLE
Js improvements

### LINKED ISSUE #273: Move comment hash JS to Timber js file
Remove the inline JS that changes the hash when a comment is posted and move it to an external sheet.

As per @mpiotrowicz and @stevebosworth's suggestion:

>I'd argue strongly that JS should be kept in JS files. Ideally, inline JS should be limited to setting liquid->dependent JS vars or calling functions

>eg this would call

>timber.updateHash(#comments)

>In this specific case as well, one thing to consider is adding in nicer keyboard support to these events too. For example right now if that page were to load with a form error, you'd be presented back to the form, but on hitting "tab", you'd be brought back to the top of the page, cycling through the main nav. By moving the JS inside the file, you'd be able to write a more robust function that also moved focus on the page to where that # is as well

### LINKED ISSUE #281: Add Fastclick
Remove the 300ms delay on mobile with [FastClick](https://github.com/ftlabs/fastclick)

### PR DESCRIPTION
**Pull request by cshold**
Fixes #281 and fixes #273

- Adds fastclick
- Move customer js to the bottom of the page
- Move comment hash JS helpers to Timber's `shop.js`
- Move login/recover password JS to Timber's `shop.js`

Demo shop for new updates - http://steve-dev.myshopify.com/

@stevebosworth @mpiotrowicz

### CODE REVIEW

**Comment by cshold** [comment]
Any final comments on this PR @mpiotrowicz @stevebosworth?

File: assets/shop.js.liquid [hunk 1]
```diff
@@ -116,6 +124,14 @@ timber.accessibleNav = function () {
   }
 };
 
+timber.getHash = function () {
+  return window.location.hash;
+};
+
+timber.updateHash = function (hash) {
+  window.location.hash = '#' + hash;
```
Comment (mpiotrowicz) [inline assets/shop.js.liquid]: how did you end up handling this case?

File: layout/theme.liquid [hunk 2]
```diff
@@ -163,41 +152,62 @@
...
+        });
+      });
+    </script>
   {% endif %}
 
+  {% comment %}
+    If you store has customer accounts disabled, you can remove the following JS file
+  {% endcomment %}
+  {% if template contains 'customers' %}
+    {{ 'shopify_common.js' | shopify_asset_url | script_tag }}
```
Comment (mpiotrowicz) [inline layout/theme.liquid]: is this ok at the bottom of the page? Just want to confirm

File: layout/theme.liquid [hunk 3]
```diff
@@ -163,41 +152,62 @@
 
   {% include 'footer' %}
 
-  {% comment %}
-    Helper scripts that you can extend as needed.
-  {% endcomment %}
   {{ 'shop.js' | asset_url | script_tag }}
```
Comment (mpiotrowicz) [inline layout/theme.liquid]: is this no longer the case with shop.js? (ie. the comments above)

File: layout/theme.liquid [hunk 4]
```diff
@@ -45,17 +45,6 @@
...
 
-  {% comment %}
-    If you store has customers disabled, you can remove the following JS file
-  {% endcomment %}
-  {% if template contains 'customers' %}
-    {{ 'shopify_common.js' | shopify_asset_url | script_tag }}
-  {% endif %}
-
-  {% comment %}
-    Theme dependent JS is loaded at the very bottom of this file for performance reasons.
```
Comment (mpiotrowicz) [inline layout/theme.liquid]: is this no longer the case?

**Comment by stevebosworth** [comment]
@mpiotrowicz's comments aside, LGTM :+1:

**Pull request merged**